music_assistant/providers/lastfm_scrobble/__init__.py

```diff
@@ -123,12 +135,12 @@ def scrobble() -> None:
         if self.should_scrobble(report):
             await asyncio.to_thread(scrobble)
 
-        if report.fully_played:
-            # reset currently playing to avoid it expiring when looping songs
-            self._currently_playing = None
-
     def should_scrobble(self, report: MediaItemPlaybackProgressReport) -> bool:
         """Determine if a track should be scrobbled, to be extended later."""
+        if self._last_scrobbled == report.uri:
+            self.logger.debug("skipped scrobbling due to duplicate event")
+            return False
+
```

Comment:
If you want we can also simply implement this check in the queue controller to prevent the duplicate event in the first place

Reply:
I'm planning to take a in-depth look in how that flow now works to see how that can be improved, like a clearer distinction in the different triggers for the event for example (`state change` vs `item change` vs `30s interval`).
The duplicate itself isn't a problem imo, it's not even a real duplicate, it's just tricky to recognize at the moment,

Reply:
OK, I'll leave it with you then. the logic to seek for is the method "_handle_playback_progress_report" within the player_queues controller.